juju/juju
# Support a simplifed dump-model

## Description of change

juju status is shown to be terribly slow on very large systems. This is mostly due to how complexity of status has grown over time and the O(n²) operations it has. This code adds an arg struct to state.Export to allow a simplified export which could be used in the status generation. This allows us to test the timing of the information gathering aspect of what status could be in order to determine if the work is worthwhile doing.

## QA steps

juju bootstrap lxd testing
juju deploy ubuntu
JUJU_DEV_FEATURE_FLAGS=developer-mode juju dump-model

observe that status history, ssh host keys, model config is in the output

JUJU_DEV_FEATURE_FLAGS=developer-mode juju dump-model --simplified

observe that those values are empty.

## Documentation changes

None at this stage.
